File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    function dfs(arr, root, level){
        if (root==null){
            return ;
        }
        if (level>arr.length){
            arr.push(root.val)
        }
        // right call
        dfs(arr, root.right, level+1);
         // left call
        dfs(arr, root.left, level+1);


    }
    if (root==null){
        return []

    }
    const arr=[]
    var level=1;
    dfs(arr, root, level)
    return arr;
    
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    void solve(TreeNode* root, vector<int>& result, int index) {
        if (root == NULL) {
            return;
        }
        // do normal pre order but do it in a way first right and then left
        if(index>=result.size()){
            result.push_back(root->val);  // pre order ka first portion
        }

        solve(root->right, result, index + 1);
        solve(root->left, result, index + 1);
    }
    vector<int> rightSideView(TreeNode* root) {

        vector<int> result;

        int index = 0;

        solve(root, result, index);

        return result;
    }
};